From 811a7313fca635f8195d007243eda431ef29e705 Mon Sep 17 00:00:00 2001
From: Weiwei Wang <weiwei.wang@windriver.com>
Date: Thu, 30 Jun 2011 12:33:01 +0800
Subject: [PATCH 220/233] Q/BMAN: set IRQF_NOBALANCING for the portal irq handler

The Q/BMAN portal is bound to a specific cpu. It make no sense
to migrate the portal irq to orther cpu when a cpu is down. And the
portal irq handler also doesn't support reentry. That mean we can't
handle two portal irq on a single cpu. So set the portal irq with
IQF_NOBALANCING flag to avoid irq migration or setting affinity
by userspace.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
[rebase it onto QorIQ-DPAA-SDK-V1-20110609-systembuilder.iso]
Integrated-by: Weiwei Wang <weiwei.wang@windriver.com>
---
 drivers/staging/fsl_qbman/bman_high.c |    2 +-
 drivers/staging/fsl_qbman/qman_high.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_qbman/bman_high.c b/drivers/staging/fsl_qbman/bman_high.c
index e085751..a1abc97 100644
--- a/drivers/staging/fsl_qbman/bman_high.c
+++ b/drivers/staging/fsl_qbman/bman_high.c
@@ -300,7 +300,7 @@ struct bman_portal *bman_create_affine_portal(
 	bm_isr_status_clear(__p, 0xffffffff);
 #ifdef CONFIG_FSL_DPA_HAVE_IRQ
 	snprintf(portal->irqname, MAX_IRQNAME, IRQNAME, config->cpu);
-	if (request_irq(config->irq, portal_isr, 0, portal->irqname, portal)) {
+	if (request_irq(config->irq, portal_isr, IRQF_NOBALANCING, portal->irqname, portal)) {
 		pr_err("request_irq() failed\n");
 		goto fail_irq;
 	}
diff --git a/drivers/staging/fsl_qbman/qman_high.c b/drivers/staging/fsl_qbman/qman_high.c
index 8f2cbc9..be85234 100644
--- a/drivers/staging/fsl_qbman/qman_high.c
+++ b/drivers/staging/fsl_qbman/qman_high.c
@@ -581,7 +581,7 @@ drain_loop:
 	qm_isr_status_clear(__p, 0xffffffff);
 #ifdef CONFIG_FSL_DPA_HAVE_IRQ
 	snprintf(portal->irqname, MAX_IRQNAME, IRQNAME, config->cpu);
-	if (request_irq(config->irq, portal_isr, 0, portal->irqname, portal)) {
+	if (request_irq(config->irq, portal_isr, IRQF_NOBALANCING, portal->irqname, portal)) {
 		pr_err("request_irq() failed\n");
 		goto fail_irq;
 	}
-- 
1.7.0.4

